File: backend/app/core/audit_closure.py
AUDIT_STATUS_CLEAN = "clean"
AUDIT_STATUS_INCOMPLETE = "incomplete"
AUDIT_STATUS_NEEDS_REVIEW = "needs_review"
AUDIT_STATUS_RELEVANT_ISSUE = "relevant_issue"

AUDIT_STATUS_LABELS = {
    AUDIT_STATUS_CLEAN: "Sem pontos relevantes",
    AUDIT_STATUS_INCOMPLETE: "Análise incompleta por falta de evidência",
    AUDIT_STATUS_NEEDS_REVIEW: "Com pontos para revisar",
    AUDIT_STATUS_RELEVANT_ISSUE: "Com pontos para verificar",
}

AUDIT_STATUS_OPTIONS = (
    AUDIT_STATUS_CLEAN,
    AUDIT_STATUS_NEEDS_REVIEW,
    AUDIT_STATUS_RELEVANT_ISSUE,
    AUDIT_STATUS_INCOMPLETE,
)


def normalize_audit_status(value: str) -> str:
    normalized = value.strip()
    if normalized not in AUDIT_STATUS_OPTIONS:
        allowed = ", ".join(AUDIT_STATUS_OPTIONS)
        raise ValueError(f"final_status_code must be one of: {allowed}")
    return normalized


def get_audit_status_label(value: str) -> str:
    return AUDIT_STATUS_LABELS.get(value, value)

File: frontend/src/lib/analysis/audit-closure.ts
export const AUDIT_CLOSURE_OPTIONS = [
  {
    description: "Pacote sem ponto relevante ativo no fechamento humano.",
    label: "Sem pontos relevantes",
    value: "clean",
  },
  {
    description: "Pacote pode ser encerrado, mas com pontos formais para revisar.",
    label: "Com pontos para revisar",
    value: "needs_review",
  },
  {
    description: "Encerramento com ponto relevante registrado formalmente.",
    label: "Com pontos para verificar",
    value: "relevant_issue",
  },
  {
    description: "Encerramento bloqueado por falta de evidência suficiente.",
    label: "Análise incompleta por falta de evidência",
    value: "incomplete",
  },
] as const;

export type AuditClosureCode =
  (typeof AUDIT_CLOSURE_OPTIONS)[number]["value"];

const auditClosureLabelMap: Record<AuditClosureCode, string> = {
  clean: "Sem pontos relevantes",
  incomplete: "Análise incompleta por falta de evidência",
  needs_review: "Com pontos para revisar",
  relevant_issue: "Com pontos para verificar",
};

export function getAuditClosureLabel(value: string) {
  return auditClosureLabelMap[value as AuditClosureCode] ?? value;
}
